src/legacy/ui/public/agg_types/controls/size.test.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { mountWithIntl, shallowWithIntl } from 'test_utils/enzyme_helpers';
+import { EuiIconTip } from '@elastic/eui';
+import { SizeParamEditor } from './size';
+
+describe('SizeParamEditor', () => {
+  let defaultProps: any;
+
+  beforeEach(() => {
+    defaultProps = {
+      value: '',
+      showValidation: false,
+      setValue: jest.fn(),
+      setValidity: jest.fn(),
+      setTouched: jest.fn(),
+    };
+  });
+
+  it('should init with the default set of props', () => {
+    const comp = shallowWithIntl(<SizeParamEditor {...defaultProps} />);
+
+    expect(comp).toMatchSnapshot();
+  });
+
+  it('should render an iconTip in the label if it was passed', () => {
+    const iconTip = <EuiIconTip position="right" content={'test'} type="questionInCircle" />;
+    const comp = shallowWithIntl(<SizeParamEditor {...defaultProps} iconTip={iconTip} />);
+
+    expect(comp).toMatchSnapshot();
```

Comment:
nit: For things like this it's safer IMHO to validate in code and not with a snapshot like you did in https://github.com/elastic/kibana/pull/41244/files#diff-5018878a1f6aa4bf3f440a1154e3d48bR88

Reply:
done.
But actually I don't see any advantage over a snapshot 😕

Reply:
It's a small thing. I have the following scenario in mind:

Someone changes the jsx structure of the size param editor and introduces a bug which causes the icon tip to not show up correctly. Jest will report two outdated snapshots and the developer will think "that makes sense, I just changed the structure", and will press `u` which greens both tests and the bug goes unnoticed (and even worse, there is now a test which checks for the wrong behavior).

In your new test this hypothetical developer will _have_ to look at the test and will definitely notice that the icon isn't rendered correctly anymore while trying to get the test green - it's much harder to accidentally render this kind of test invalid.

But definitely nothing important :)

Reply:
This make sense 👍
But if any developer would update a snapshot without taking a look inside, there should be a question to the developer 😜